lib/opam_build.ml

```diff
@@ -58,6 +58,11 @@ let pin_opam_files groups =
       )
   )
 
+let packages_of_opam_files x =
+  group_opam_files x
```

Comment:
What about moving the logic that find the package names (from `group_opam_files`) to `Analysis` ?
This logic can benefit from having its own interface and unit tests and to be reused.

Reply:
I don't think it's worthwhile right now: it's not part of the analysis step, and it's still possible to unit-test it from `Opam_build`.

Reply:
`Analysis` is not the analysis step but its result. But yes, it would be the first function of its kind there.
This function can only become more complex over time (especially `group_opam_files`), the comments in its implementation says that.
What about making it a field of `Analysis.t` ? For now, it can be computed outside of analysis but this might change as it get generalized.
My point is that this logic needs a proper abstraction.

Reply:
As discussed in person, I don't think this is worth worrying about until either (a) we add unit tests or (b) the function becomes more complex.